editoast/editoast_derive/src/snapshots/editoast_derive__model__tests__construction.snap

```diff
@@ -0,0 +1,1246 @@
+---
+source: editoast_derive/src/model.rs
+---
+// Source
+// -----------------------------------------------------------------------------
+
+#[derive(Clone, Model)]
+#[model(table = editoast_models::tables::osrd_infra_document)]
+#[model(row(type_name = "DocumentRow", derive(Debug)))]
+#[model(changeset(type_name = "DocumentChangeset", public, derive(Debug)))]
+#[model(gen(ops = crud, batch_ops = crud, list))]
+struct Document {
+    #[model(column = "id", preferred, primary)]
+    id_: i64,
+    #[model(identifier, json)]
+    content_type: String,
+    data: Vec<u8>,
+}
+
+// Macro expansion
+// -----------------------------------------------------------------------------
+
+#[automatically_derived]
+impl crate::models::Model for Document {
+    type Row = DocumentRow;
+    type Changeset = DocumentChangeset;
+    type Table = editoast_models::tables::osrd_infra_document::table;
+}
+paste::paste! {
+    impl Document { pub const [< id_ : snake : upper >] : crate ::models::ModelField <
+    Document, i64, id > = crate ::models::ModelField::new(); pub const [< content_type :
+    snake : upper >] : crate ::models::ModelField < Document, String,
+    editoast_models::tables::osrd_infra_document::content_type > = crate
+    ::models::ModelField::new(); pub const [< data : snake : upper >] : crate
+    ::models::ModelField < Document, Vec < u8 >,
+    editoast_models::tables::osrd_infra_document::data > = crate
+    ::models::ModelField::new(); }
+}
```

Comment:
Interesting, `prettyplease` completely gave up on this... Macros really gives a hard time to the tooling.

Reply:
Yep, I had spotted this one as well... The case conversion could be done in the `Model` macro. I didn't want to add too much to this PR (the `chunked_for_libpq!` was really bad, that's why I fixed it here). Follow up PR if any?